modules/post/windows/manage/killav.rb

```diff
@@ -37,41 +41,41 @@ def initialize(info = {})
     )
   end
 
-  def skip_process_name?(process_name)
-    [
+  def run
+    avs = ::File.read(
+      ::File.join(
+        Msf::Config.data_directory,
+        'wordlists',
+        'av_hips_executables.txt'
+      )
+    )
+    avs = avs.strip.downcase.split("\n").uniq
+
+    skip_processes = [
       '[system process]',
       'system'
-    ].include?(process_name)
-  end
+    ]
 
-  def run
-    avs = ::File.read(::File.join(Msf::Config.data_directory, 'wordlists',
-                                  'av_hips_executables.txt')).strip
-    avs = Set.new(avs.split("\n"))
+    av_processes = get_processes.reject { |p| skip_processes.include?(p['name'].downcase) }.keep_if { |p| avs.include?(p['name'.downcase]) }
+    if av_processes == 0
```

Comment:
Is this meant to be `av_processes.length`? Seems like this is an array or list of some sort, not a number given the later call to `av_processes.each`.

Reply:
```
    61: 
    62:     av_processes = get_processes.reject { |p| skip_processes.include?(p['name'].downcase) }.keep_if { |p| avs.include?(p['name'.downcase]) }
 => 63:     if av_processes == 0
    64:       print_status('No target processes were found.')
    65:       return
    66:     end
    67: 
    68:     processes_killed = 0

[2] pry(#<Msf::Modules::Post__Windows__Manage__Killav::MetasploitModule>)> av_processes
=> []
[3] pry(#<Msf::Modules::Post__Windows__Manage__Killav::MetasploitModule>)> av_processes == 0
=> false
[4] pry(#<Msf::Modules::Post__Windows__Manage__Killav::MetasploitModule>)> av_processes.length == 0
=> true
[5] pry(#<Msf::Modules::Post__Windows__Manage__Killav::MetasploitModule>)> 
```

Looks like this should be changed to `av_processes.length`. Will make that change as part of testing and landing this.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/commit/9a8afda0a2a40e2352f2a3e131067db707bd0040